### PR TITLE
openh264/* upgrade to conan v2

### PR DESCRIPTION
**Pull request by Nomalah**
Specify library name and version:  **openh264/\***

Upgrade openh264 to conan v2

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Nomalah** [comment]
(also added v2.3.1)

File: recipes/openh264/all/conanfile.py [hunk 1]
```diff
@@ -24,73 +43,72 @@ class OpenH264Conan(ConanFile):
...
+            except Exception:
+                pass
+    
+    def layout(self):
+        # src_folder must use the same source folder name the project
+        basic_layout(self, src_folder="src")
 
     def build_requirements(self):
         if self.settings.arch in ("x86", "x86_64"):
             self.build_requires("nasm/2.15.05")
```
Comment (SpaceIm) [inline recipes/openh264/all/conanfile.py:73]:
```suggestion
            self.tool_requires("nasm/2.15.05")
```

File: recipes/openh264/all/conanfile.py [hunk 2]
```diff
@@ -24,73 +43,72 @@ class OpenH264Conan(ConanFile):
...
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+    
+    def layout(self):
+        # src_folder must use the same source folder name the project
```
Comment (SpaceIm) [inline recipes/openh264/all/conanfile.py:68]:
```suggestion
```

File: recipes/openh264/all/test_v1_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    def build_requirements(self):
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
+            # Workaround for CMake bug with error message:
+            # Attempting to use @rpath without CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG being
+            # set. This could be because you are using a Mac OS X version less than 10.5
+            # or because CMake's platform configuration is corrupt.
+            # FIXME: Remove once CMake on macOS/M1 CI runners is upgraded.
+            self.build_requires("cmake/3.22.0")
+
```
Comment (SpaceIm) [inline recipes/openh264/all/test_v1_package/conanfile.py:17]:
```suggestion
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/openh264/all/test_v1_package/conanfile.py (hunk 3 above)
Comment (SpaceIm) [reply recipes/openh264/all/test_v1_package/conanfile.py:17]: not required anymore

File: recipes/openh264/all/conandata.yml [hunk 4]
```diff
@@ -1,26 +1,38 @@
...
   "2.1.1":
     url: "https://github.com/cisco/openh264/archive/v2.1.1.tar.gz"
     sha256: "af173e90fce65f80722fa894e1af0d6b07572292e76de7b65273df4c0a8be678"
   "1.7.0":
     url: "https://github.com/cisco/openh264/archive/v1.7.0.tar.gz"
     sha256: "9c07c38d7de00046c9c52b12c76a2af7648b70d05bd5460c8b67f6895738653f"
 patches:
+  "2.3.1":
+    - patch_file: "patches/2.3.1-0001-platform-android.mk.patch"
+      patch_type: "conan"
```
Comment (prince-chrismc) [inline recipes/openh264/all/conandata.yml:14]: Would portability be a better fit?

Maybe set these to upstream so we don't have to keep applying it?

File: recipes/openh264/all/conandata.yml [hunk 5]
```diff
@@ -1,26 +1,38 @@
...
+      patch_description: "Android Fix"
     - patch_file: "patches/2.1.1-0002-macos-relocatable-shared.patch"
-      base_path: "source_subfolder"
-    - patch_file: "patches/2.1.1-0003-no-fpic.patch"
-      base_path: "source_subfolder"
+      patch_type: "conan"
+      patch_description: "Macos relocatable shared fix"
     - patch_file: "patches/1.7.0-0004-mingw-override-CC-CXX-AR-from-env.patch"
-      base_path: "source_subfolder"
+      patch_type: "conan"
```
Comment (prince-chrismc) [inline recipes/openh264/all/conandata.yml:27]: Sorry for being picky but id like to make sure these are correct so we can measure the types and reasons 

https://github.com/conan-io/conan-center-index/blob/master/docs/conandata_yml_format.md#patch_type

```suggestion
      patch_type: "portability"
```

Would be a better fit based on  the description

**Review by prince-chrismc — COMMENTED** [review]
Thank you so much for adding the patch fields just would like to change them to match the docs.

I believe they are all about fixes for different system and they should be type portability 

Please let me know if that's not correct:)

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/openh264/all/conanfile.py [hunk 6]
```diff
@@ -1,8 +1,29 @@
...
+    libcxx = conanfile.settings.get_safe("compiler.libcxx")
+    if libcxx in ["libstdc++", "libstdc++11"]:
+        return "stdc++"
+    elif libcxx in ["libc++"]:
+        return "c++"
+    elif libcxx in ["c++_shared"]:
+        return "c++_shared"
+    elif libcxx in ["c++_static"]:
+        return "c++_static"
+    return None
```
Comment (uilianries) [inline recipes/openh264/all/conanfile.py:23]:
```suggestion
```

Please, use https://docs.conan.io/en/latest/reference/tools.html#tools-stdcpp-library instead and add a `TODO` to be updated after 1.54.0. Try avoiding duplicated code from Conan tools.

File: recipes/openh264/all/conanfile.py [hunk 7]
```diff
@@ -24,73 +45,72 @@ class OpenH264Conan(ConanFile):
...
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
```
Comment (MartinDelille) [inline recipes/openh264/all/conanfile.py:67]: Since 1.52.0, you can use `self.options.rm_safe("fPIC")` to avoid this warning.

**Review by MartinDelille — CHANGES_REQUESTED** [review]

**Review by MartinDelille — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**